tests/unit_tests/test_resample_cube_temporal.py

```diff
@@ -0,0 +1,974 @@
+            ],
+            "t",
+            None,
+            "DimensionMismatch: The temporal dimensions for resampling don't match.",
+        ),
+        (  # dimension is given, but not temporal in data and target
+            [1],
+            [1],
+            [
+                {"labels": ["2022-03-21"], "name": "t", "type": "other"},
+                {"labels": ["B01"], "name": "b", "type": "bands"},
+            ],
+            [1],
+            [1],
+            [
+                {"labels": ["2022-03-21"], "name": "t", "type": "other"},
+                {"labels": ["B01"], "name": "b", "type": "bands"},
+            ],
+            "t",
+            None,
+            "DimensionMismatch: The temporal dimensions for resampling don't match.",
+        ),
+        (  # dimension not given, no temporal dimensions match by name in data and target
+            [1],
+            [1],
+            [
+                {"labels": ["2022-03-21"], "name": "t1", "type": "other"},
+                {"labels": ["B01"], "name": "b", "type": "bands"},
+            ],
+            [1],
+            [1],
+            [
+                {"labels": ["2022-03-21"], "name": "t2", "type": "other"},
+                {"labels": ["B01"], "name": "b", "type": "bands"},
+            ],
+            None,
+            None,
+            "DimensionMismatch: The temporal dimensions for resampling don't match.",
+        ),
+    ],
+)
+def test_mask_exceptions(
+    resample_cube_temporal_process_code,
+    construct_datacube,
+    data_array,
+    data_shape,
+    data_dimensions,
+    target_array,
+    target_shape,
+    target_dimensions,
+    dimension,
+    valid_within,
+    error_message,
+):
+
+    additional_js_code_to_run = load_datacube_code()
+    arguments = ""
+
+    if data_array and data_shape and data_dimensions:
+        data_cube = construct_datacube(
+            "dataCube", data_array, data_shape, data_dimensions
+        )
+        additional_js_code_to_run = additional_js_code_to_run + data_cube
+        arguments = arguments + f"'data': dataCube,"
+
+    if target_array and target_shape and target_dimensions:
+        target_cube = construct_datacube(
+            "targetCube", target_array, target_shape, target_dimensions
+        )
+        additional_js_code_to_run = additional_js_code_to_run + target_cube
+        arguments = arguments + f"'target': targetCube,"
+
+    if dimension:
+        additional_js_code_to_run = (
+            additional_js_code_to_run + f"const dimension = {json.dumps(dimension)};"
+        )
+        arguments = arguments + f"'dimension': {json.dumps(dimension)},"
+
+    if valid_within:
+        additional_js_code_to_run = (
+            additional_js_code_to_run
+            + f"const valid_within = {json.dumps(valid_within)};"
+        )
+        arguments = arguments + f"'valid_within': {json.dumps(valid_within)}"
+
+    process_arguments = f"{{" + arguments + f"}}"
+
+    print(arguments)
```

Comment:
Thanks, removed. I also removed the tests for `valid_within` from `unit_tests/test_resample_cube_temporal.py` and added one more to the tests in `integration_tests/test_integration.py`

Reply:
>  I also removed the tests for valid_within from unit_tests/test_resample_cube_temporal.py and added one more to the tests in integration_tests/test_integration.py

It was fine as it was, you can add the unit tests back.
```python
# should be in tests/integration_tests/test_integration.py
# because the process needs is_valid process
# currently solved by naively loading the additional code for the dependency process
````
There is no other way for setting up unit tests other the way it was done, there's no issue with it. I didn't notice these comments previously.

Reply:
Thanks. Readded.